File: JY/10월 4주차/H-Index.py
MAX_COUNT = 10001
def solution(citations):
    answer = 0
    #STEP1. h=1부터 10000 
    for h in range(1, MAX_COUNT):
        #STEP2. N만큼의 논문 순회(1000)
        cnt = 0
        for i in range(len(citations)):
            if citations[i] >= h:
                cnt+=1
        if cnt >= h :
            answer = h
        #-- 안되는 순간 나머지는 다 안됨
        else:
            break
    #STEP3. 
    return answer

File: BW/10월3주차/h-index.js
function solution(citations) {
    var answer = citations;
    answer.sort((a,b)=>b-a); // 내림차순
    
    for(var i = 0;i<answer.length;i++){
        if(i>=answer[i]) return i;
    }

    // 제한에 걸리지 않은 경우 모두 출력
    return answer.length;
}